node-scripts/spec/pipeline3/impl/dockerCompose.spec.js

```diff
@@ -0,0 +1,165 @@
+  const dockerCompose = require(path.resolve(
+    "src/pipeline3/impl/dockerCompose"
+  ));
+
+  const scripts = require(path.resolve(
+    "src/" + config.getJobNameForPipeline3() + "/scripts"
+  ));
+
+  describe("Pipeline steps", function() {
+    var instanceDef = {
+      uuid: "cacb5448-46b0-4808-980d-5521775671c0",
+      name: "cambodia1",
+      type: "dev",
+      group: "tlc",
+      deployment: {
+        hostDir: "/var/docker-volumes/",
+        type: "dockerCompose",
+        value: {
+          image: "mekomsolutions/bahmni",
+          tag: "cambodia-release-0.90",
+          ports: {
+            "443": "8733",
+            "80": "8180"
+          },
+          networks: ["network1", "network2"],
+          services: ["proxy", "openmrs", "mysql"]
+        },
+        host: {
+          type: "ssh",
+          value: {
+            ip: "hsc-dev.mekomsolutions.net",
+            user: "mekom",
+            port: "22"
+          }
+        }
+      }
+    };
+
+    it("should generate docker compose Pre-Host Preparation deployment script", () => {
+      expect(
+        dockerCompose.preHostPreparation.getDeploymentScript(instanceDef)
+      ).toEqual("");
+    });
+
+    it("should generate docker compose Pre-Host Preparation data script", () => {
+      expect(
+        dockerCompose.preHostPreparation.getDataScript(instanceDef)
+      ).toEqual("");
+    });
+
+    it("should generate docker compose Pre-Host Preparation artifacts script", () => {
+      expect(
+        dockerCompose.preHostPreparation.getArtifactsScript(instanceDef)
+      ).toEqual("");
+    });
+
+    it("should generate docker compose Host Preparation deployment script", () => {
+      let expected =
+        scripts.remote(
+          instanceDef.deployment.host.value,
+          scripts.gitClone(
+            instanceDef.deployment.value.gitUrl,
+            path
+              .resolve(
+                instanceDef.deployment.hostDir,
+                instanceDef.name,
+                "bahmni_docker"
+              )
+              .toString(),
+            instanceDef.deployment.value.commitId
+          )
+        ) +
+        "\n" +
+        scripts.remote(
+          instanceDef.deployment.host.value,
+          scripts.createEnvVarFile(instanceDef)
+        ) +
+        "\n" +
+        scripts.remote(
+          instanceDef.deployment.host.value,
+          "cd " +
+            path
+              .resolve(
+                instanceDef.deployment.hostDir,
+                instanceDef.name,
+                "bahmni_docker"
+              )
+              .toString() +
+            " && docker-compose -p " +
+            instanceDef.name +
+            " --env-file=" +
+            path
+              .resolve(
+                instanceDef.deployment.hostDir,
+                instanceDef.name,
+                instanceDef.name + ".env"
+              )
+              .toString() +
+            " build --pull proxy openmrs mysql" +
```

Comment:
This shouldn't be tested that way.
We should call the method that builds up the service list. (`getInstanceServices()`)

And then the method `getInstanceServices()` should itself be unit tested.